devito/ops/operator.py

```diff
@@ -108,13 +109,12 @@ def _specialize_iet(self, iet, **kwargs):
 
     @property
     def hcode(self):
-        return ''.join(str(kernel) for kernel in self._ops_kernels)
+        return ''.join(str(kernel.root) for kernel in self._func_table.values())
 
     def _finalize(self, iet, parameters):
-        # Applies _finalize for each generated OPS kernel, which will generate the .h file
-        self._ops_kernels = [super(OperatorOPS, self)._finalize(
-            kernel, derive_parameters(kernel, True))
-            for kernel in self._ops_kernels]
+
+        self._func_table.update({k: MetaCall(super(OperatorOPS, self)._finalize(v.root, derive_parameters(v.root, True)), v.local) # noqa
```

Comment:
I accept suggestion for this line here... I tried to separate but it got weird and confusing.

Reply:
```
for k, v in list(self._func_table.items()):
    parameters = derive_parameters(v.root, True)
    root = super(OperatorOPS, self)._finalize(v.root, parameters)
    self._func_table[k] = MetaCall(root, v.local)
```
?

Reply:
I was only thinking about list comprehension, but this seems better.